blog_post.md

```diff
@@ -190,54 +190,55 @@ We use the following hyperparameter settings in all our experiments:
-slowing down divergence, whereas employing a separate target network is more
-effective.
-We also observe that when having both tricks,the network's divergence is slowed
-down even more.
-
-Based on all our experiments we can say that divergence with DQN is
-unavoidable, however it can be slowed down, allowing the network to learn more,
-and achieve a higher score.
-We can see this in the following plots, which show us the distribution of
-obtained rewards.
-
-![MountainCar reward]({{page.img_dir}}violinplot_reward_MountainCar-v0_0.99.png)
-![Acrobot reward]({{page.img_dir}}violinplot_reward_Acrobot-v0_0.99.png)
-
-In the case of having both memory sampling and a target network enabled, we
-achieved the highest reward.
-However, the graphs also tell us that although the target network is more
-effective at slowing down divergence, it also negatively influences DQN's
-learning capabilities.
-
-<!--- // TOOD: maybe think of a theoretical explanation of this? -->
-
-Lastly, we look at our results on the CartPole environment, for which we
-explicitly disabled the error clipping.
-Mnih et. al. argued that this improved the stability of the neural network,
-however based on our findings, we can say that it also significantly reduced
-the network's learning capacity (and the reward obtained).
-
-![CartPole divergence]({{page.img_dir}}violinplot_q_divergence_CartPole-v0_0.99.jpeg)
-<!--- <> TODO: add reward image of CartPole. -->
-
-We see that error clipping is instrumental to making the tricks work.
-Without error clipping, the tricks actually hasten the divergence, rather than
-slow it down in our experiments.
+## Results
+
+The results of the best performing hyper-parameters are shown in the below figure.
+The figure displays a scatter plot of DQN's performance in the last 20 episodes of our experiments.
+We use the x-axis to measure the max-|Q|, identifying divergence (log-scale in the Acrobot experiment), and on the y-axis we plot the average *return*.
+The figure allows us to analyse the effects of the employed tricks not only on the max-|Q|, but also on the overall performance of the DQN agent.
+We shall note the obtained results first for each environment separately, and then we shall point out our general conclusions.
+
+![DQN reward-divergence scatter]({{page.img_dir}}scatter.png)
+<!-- TODO: add titles to the plots, identifying the experiment -->
+<!-- TODO: rename 'Average Rewards per Episode' to 'Average Return' -->
+<!-- TODO: rename 'Maximum Q values' to 'Maximum Absolute Q value' -->
+
+To begin with, let us look at the Mountain Car results.
+It is obvious that vanilla DQN diverges and miserably fails at learning a good estimate of the Q-value function.
+Similarly, the network with Memory Replay also performs badly, with the exception of a few outliers, which perform quite well.
+We can say that when the Memory Replay network did not diverge, it obtained a very good overall return.
```

Comment:
Too many "we can say that..." . In this example, it can be avoided. We also need to note the significance of what we're seeing.

```suggestion
When the Memory Replay network did not diverge, it obtained a good overall return. This is an interesting observation, as it suggests that our measure of divergence is indeed predictive of final performance.   
```

Reply:
Yeah, I agree with the "we can say that" thing.
I tried to use different linking words wherever possible.